### PR TITLE
Add support for --external to esbuild configuration

### PR DESCRIPTION
**Pull request by chris-armstrong**
This PR adds support to the esbuild `BuildProperties` to specify a set of externals to use when building a source code bundle. This is useful when the package you require is on a layer and cannot be bundled with esbuild. It converts an array of `Externals` to a set of `esbuild` `--external:<module>` parameters.

For example, if your `BuildProperties` contains:

```yaml
  MyFunction:
    Metadata:
      BuildProperties:
        Externals: ['aws-sdk', 'puppeteer']
```

Then `--external:aws-sdk --external:puppeteer` will be added to the esbuild command line.

(Please forgive any non-idiomatic Python code - this is the first time writing Python for a number of years so I'm rusty on latest best practice).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws_lambda_builders/workflows/nodejs_npm_esbuild/actions.py [hunk 1]
```diff
@@ -71,38 +74,26 @@ def execute(self):
...
         args = explicit_entry_points + ["--bundle", "--platform=node", "--format=cjs"]
         minify = self.bundler_config.get("minify", True)
         sourcemap = self.bundler_config.get("sourcemap", True)
         target = self.bundler_config.get("target", "es2020")
+        externals = self.bundler_config.get("externals", [])
         if minify:
             args.append("--minify")
         if sourcemap:
             args.append("--sourcemap")
+        if len(externals) > 0:
```
Comment (mildaniel) [inline aws_lambda_builders/workflows/nodejs_npm_esbuild/actions.py:88]: Very minor comment, but can we make this `if externals:` to be consistent with PEP8 styling guide?

**Review by mildaniel — APPROVED** [review]

**Review by mndeveci — COMMENTED** [review]
@chris-armstrong thanks for the PR! Can you address the concern from @mildaniel? Other than that, LGTM.

**Comment by chris-armstrong** [comment]
> @chris-armstrong thanks for the PR! Can you address the concern from @mildaniel? Other than that, LGTM.

The code uses ‘if externals:’ now - was there anything else?

**Review by mndeveci — APPROVED** [review]
>> @chris-armstrong thanks for the PR! Can you address the concern from @mildaniel? Other than that, LGTM.

> The code uses ‘if externals:’ now - was there anything else?

Sorry my bad!

**Comment by mndeveci** [comment]
@chris-armstrong one last thing. This PR was pointing to master branch. I now changed it to develop, but it is out of date now. Can you re-base your changes from develop so that we can merge into?

**Comment by chris-armstrong** [comment]
I can, but I’ll be away from my computer for the next few days so I won’t be able to rebase immediately

**Comment by chris-armstrong** [comment]
> @chris-armstrong one last thing. This PR was pointing to master branch. I now changed it to develop, but it is out of date now. Can you re-base your changes from develop so that we can merge into?
> 
> If you want, you can close this PR and open a new one, which should branch out from develop.

I've opened #353 that has been rebased against develop

**Comment by moelasmar** [comment]
@chris-armstrong .. should we close this PR?

**Comment by chris-armstrong** [comment]
> @chris-armstrong .. should we close this PR?

Yes I'll close this now

**Pull request closed without merging**